bot/exts/moderation/infraction/management.py

```diff
@@ -73,29 +73,21 @@ async def infraction_append(
         Use "p" or "permanent" to mark the infraction as permanent. Alternatively, an ISO 8601
         timestamp can be provided for the duration.
         """
-        if isinstance(infraction_id, str):
-            old_infraction = await self.get_latest_infraction(ctx.author.id)
-
-            if old_infraction is None:
-                await ctx.send(
-                    ":x: Couldn't find most recent infraction; you have never given an infraction."
-                )
-                return
-
-            infraction_id = old_infraction["id"]
-
-        else:
-            old_infraction = await self.bot.api_client.get(f"bot/infractions/{infraction_id}")
-
-        reason = fr"{old_infraction['reason']} **\|\|** {reason}"
+        if not infraction:
+            return
 
-        await self.infraction_edit(infraction_id=infraction_id, duration=duration, reason=reason)
+        await self.infraction_edit(
+            ctx=ctx,
+            infraction=infraction,
+            duration=duration,
+            reason=fr"{infraction['reason']} **\|\|** {reason}",
+        )
```

Comment:
You could have kept this as 2 lines rather than 6 (only the `reason` needs to be a kwarg). But it's fine.